src/CoolerFactory.sol

```diff
@@ -78,27 +77,33 @@ contract CoolerFactory {
     // --- EMIT EVENTS -----------------------------------------------
 
     enum Events {
-        Request,
-        Rescind,
-        Clear,
-        Repay
+        RequestLoan,
+        RescindRequest,
+        ClearRequest,
+        RepayLoan,
+        RollLoan,
+        DefaultLoan
     }
 
     /// @notice emit an event each time a request is interacted with on a Cooler.
-    /// @param id_ loan or request identifier.
-    /// @param ev_ event type.
-    /// @param amount_ to be logged by the event.
+    /// @param  id_ loan or request identifier.
+    /// @param  ev_ event type.
+    /// @param  amount_ to be logged by the event.
     function newEvent(uint256 id_, Events ev_, uint256 amount_) external {
         require(created[msg.sender], "Only Created");
 
-        if (ev_ == Events.Clear) {
-            emit Clear(msg.sender, id_);
-        } else if (ev_ == Events.Repay) {
-            emit Repay(msg.sender, id_, amount_);
-        } else if (ev_ == Events.Rescind) {
-            emit Rescind(msg.sender, id_);
-        } else if (ev_ == Events.Request) {
-            emit Request(msg.sender, address(Cooler(msg.sender).collateral()), address(Cooler(msg.sender).debt()), id_);
+        if (ev_ == Events.RequestLoan) {
+            emit RequestLoan(msg.sender, address(Cooler(msg.sender).collateral()), address(Cooler(msg.sender).debt()), id_);
+        } else if (ev_ == Events.RescindRequest) {
+            emit RescindRequest(msg.sender, id_);
+        } else if (ev_ == Events.ClearRequest) {
+            emit ClearRequest(msg.sender, id_);
+        } else if (ev_ == Events.RepayLoan) {
+            emit RepayLoan(msg.sender, id_, amount_);
+        } else if (ev_ == Events.RollLoan) {
+            emit RollLoan(msg.sender, id_);
+        } else {
```

Comment:
I would have an explicit check for the event type == rollover, and in the else case, raise an error.